Bottom: e2d86c570cdcf4687fc5f27cd935d818f4a8da80
Top:    56cf76b9a3fd13e99f340cce9cea5a518b63b0fd
Author: Jason Wang <jasowang@redhat.com>
Date:   2016-05-12 12:05:21 +0800

Refresh of tuntap-introduce-tx-skb-ring

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 7a5687f..602f45c 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -175,9 +175,11 @@ struct tun_file {
 	};
 	struct list_head next;
 	struct tun_struct *detached;
+	/* reader lock */
 	spinlock_t rlock;
 	unsigned long tail;
 	struct tun_desc tx_descs[TUN_RING_SIZE];
+	/* writer lock */
 	spinlock_t wlock;
 	unsigned long head;
 };
